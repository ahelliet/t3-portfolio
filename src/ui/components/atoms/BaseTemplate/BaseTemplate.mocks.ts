import { IBaseTemplate } from './BaseTemplate'

const base: IBaseTemplate = {
  sampleText: 'Hello world!',
}

export const mockBaseTemplateProps = {
  base,
}
