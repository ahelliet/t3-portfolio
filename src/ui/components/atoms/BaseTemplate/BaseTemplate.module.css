.BaseTemplate {
  @apply flex h-8 w-[50vw] items-center justify-center rounded-md text-blue-800 ring-[1px] ring-blue-800;
}
