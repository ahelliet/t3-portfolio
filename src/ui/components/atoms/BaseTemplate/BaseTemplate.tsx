import React from 'react'
import styles from './BaseTemplate.module.css'

export interface IBaseTemplate {
    sampleText: string
}

const BaseTemplate: React.FC<IBaseTemplate> = ({ sampleText }) => {
    return (
        <div className={styles.BaseTemplate}>{sampleText}</div>
    )
}

export default BaseTemplate;