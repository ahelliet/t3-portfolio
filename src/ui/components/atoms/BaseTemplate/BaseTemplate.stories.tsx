import { ComponentStory, ComponentMeta } from '@storybook/react'
import React from 'react'

import BaseTemplate from './BaseTemplate'
import { mockBaseTemplateProps } from './BaseTemplate.mocks'

export default {
    title: 'Template/BaseTemplate',
    component: BaseTemplate,
    args: {},
    argTypes: {},
} as ComponentMeta<typeof BaseTemplate>

const Template: ComponentStory<typeof BaseTemplate> = (args) => (
    <BaseTemplate {...args} />
)

export const Base = Template.bind({})

Base.args = { ...mockBaseTemplateProps.base }
Base.storyName = 'BaseTemplate'
