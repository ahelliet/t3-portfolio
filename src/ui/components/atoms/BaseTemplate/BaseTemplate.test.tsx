import { composeStories } from '@storybook/testing-react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'

import * as stories from './BaseTemplate.stories'

describe('components/Example', () => {
    const { Base } = composeStories(stories)

    test('Component is not null', () => {
        const { container } = render(<Base sampleText="This is a test text." />)
        expect(container).not.toBeNull()

        // Snapshot test
        expect(container).toMatchSnapshot()
    })
})
