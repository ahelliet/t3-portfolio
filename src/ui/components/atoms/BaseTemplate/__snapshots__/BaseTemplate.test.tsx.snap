// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Example Component is not null 1`] = `
<div>
  <div
    class="BaseTemplate"
  >
    This is a test text.
  </div>
</div>
`;
